tests/test_env.py

```diff
@@ -202,10 +202,14 @@ def test_process_env_install_from_index(
     assert package_name in installed
     if version is not None:
         assert installed[package_name] == version
+    # legacy method
+    env.process_env.install_from_index(
+        [Requirement.parse(package_name + (f"=={version}" if version is not None else ""))],
+        use_pip_config=True,
+    )
```

Comment:
What are you actually testing here? The package is already installed at this point, isn't it? Same question for the `install_from_source` further down.